File: payment-service-app/microservice-payment-gateway/src/main/java/com/microservice/payment/entity/Merchant.java
package com.microservice.payment.entity;

import jakarta.persistence.*;
import java.util.List;

@Entity
@Table(name = "merchant")
public class Merchant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "merchant", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Transaction> transactions;

    // Getters and Setters
}

File: payment-service-app/nodejs-framework-payment-app/express-payment/src/domain/entity/Merchant.ts
// Merchant.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToOne, OneToMany, JoinColumn, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import { User } from '../entity/User';
import { Transaction } from '../entity/Transaction';

@Entity()
export class Merchant {
    @PrimaryGeneratedColumn()
    merchant_id!: number;

    @Column({ name: 'business_name' })
    businessName!: string;

    @Column({ name: 'business_address' })
    businessAddress!: string;

    @CreateDateColumn()
    created_at!: Date;

    @UpdateDateColumn()
    updated_at!: Date;

    @OneToOne(() => User, (user: User) => user.merchant)
    @JoinColumn({ name: 'user_id' })
    user!: User;

    @OneToMany(() => Transaction, transaction => transaction.merchant)
    transactions!: Transaction[];
}